torch/distributed/nn/api/remote_module.py

```diff
@@ -164,7 +168,7 @@ def __init__(
 
         # Create the module on the remote side.
         self.module_rref = rpc.rpc_sync(
-            on, _create_module, (module_cls, args, kwargs, _module_interface_cls)
+            on, _create_module, (module_cls, args, kwargs, device, _module_interface_cls)
```

Comment:
It might be better to wrap device as `device = torch.device(device)` on the client side so that if we have an invalid device type an appropriate error is raised to the user. In the current scenario, we would invoke an RPC on the remote end and the `.to()` on the remote end would fail. The RPC framework would return that exception, but I'm not sure if it would be as clear as raising on the client side.

Reply:
Actually nvm, we should probably not do this. The reason being that sometimes a device might exist only on the remote node and not on the client and this would give us unintended errors.

Reply:
Thanks for the comments. We can always revisit the API once our use cases become more clear. I believe RPC framework now is still in experimental phase, so the API here doesn't have to be finalized.

Reply:
> Actually nvm, we should probably not do this. The reason being that sometimes a device might exist only on the remote node and not on the client and this would give us unintended errors.

Reverted this change after the second review.